### PR TITLE
Duplicate game validation

### PR DESCRIPTION
**Pull request by shadre**

### CODE REVIEW

File: app/services/league_of_legends/achievement_winners_finder.rb [hunk 1]
```diff
@@ -6,30 +6,53 @@ class AchievementWinner < Struct.new(
...
         )
       end
     end
 
     private
 
-    def winners(achievement_value_per_player)
-      biggest_achievement_value = achievement_value_per_player.values.max
+    def winners(achievement_value_per_player, limit)
+      biggest_achievement_values = achievement_value_per_player
```
Comment (ArturT) [inline app/services/league_of_legends/achievement_winners_finder.rb]: general idea: sometimes I format lines like this:

```ruby
  biggest_achievement_values =
    achievement_value_per_player
    .values
    .uniq
    .max(limit)
```

File: app/services/league_of_legends/elo_per_player.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module LeagueOfLegends
+  class EloPerPlayer
```
Comment (ArturT) [inline app/services/league_of_legends/elo_per_player.rb]: This service only takes `cutoff_date` and pass it to `Elo::RankingCalculator` service. Why not just use `Elo::RankingCalculator` directly in the places where you call now `EloPerPlayer`?

File: app/services/slack_query.rb [hunk 3]
```diff
@@ -40,4 +44,21 @@ def unique_players
       errors.add("players", "can't be duplicated")
     end
   end
+
+  def recently_unique_match
+    recently_played_matches = Match.where("created_at > ?", RECENCY_PERIOD.ago)
```
Comment (AkkarinForest) [inline app/services/slack_query.rb]: We could use a repository for that? something like: `MatchesRepository.new.fetch_matches(start_date: RECENCY_PERIOD.ago)`

Comment (shadre) [reply app/services/slack_query.rb]: Good catch, Sir. I shall change that promptly.

**Pull request merged**